File: leetcode.com/0082_remove_duplicates_from_sorted_list_2/solution.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next

class Solution:
    def deleteDuplicates(self, head: ListNode) -> ListNode:
        target = curr = prev = None

        curr = head

        while curr:
            if curr.next and curr.next.val == curr.val:
                target = curr.val

            if curr.val != target:
                prev = curr
                curr = curr.next
                continue

            while curr and curr.val == target:
                if curr == head:
                    head = curr.next
                    curr = head
                else:
                    curr = curr.next
                    prev.next = curr

        return head

File: leetcode.com/0082_remove_duplicates_from_sorted_list_2/solution.go
/**
 * Definition for singly-linked list.
 * type ListNode struct {
 *     Val int
 *     Next *ListNode
 * }
 */

 func deleteDuplicates(head *ListNode) *ListNode {
    var target = -101
    var begin, curr, prev *ListNode

    begin = head
    curr = head
    prev = nil

    for curr != nil {

        /* find duplicated value */
        if (curr.Next != nil && curr.Val == curr.Next.Val) {
            target = curr.Val
        }

        if (curr.Val != target) {
            prev = curr
            curr = curr.Next
            continue
        }

        for curr != nil && curr.Val == target {

            if (curr == begin) {
                begin = curr.Next
                curr = begin
            } else {
                curr = curr.Next
                prev.Next = curr
            }
        }
    }

    return begin
}
